modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney_controller.rb

```diff
@@ -26,65 +26,63 @@ def show
           end
 
-          submit_power_of_attorney(poa_code)
+          submit_power_of_attorney(validated_poa_code, '2122A')
         end
 
-        def submit_power_of_attorney(poa_code)
-          power_of_attorney = ClaimsApi::PowerOfAttorney.find_using_identifier_and_source(header_md5:,
-                                                                                          source_name:)
-          unless power_of_attorney&.status&.in?(%w[submitted pending])
-            attributes = {
-              status: ClaimsApi::PowerOfAttorney::PENDING,
-              auth_headers:,
-              form_data: params,
-              current_poa: current_poa_code,
-              header_md5:
-            }
-            attributes.merge!({ source_data: }) unless token.client_credentials_token?
-
-            # use .create! so we don't need to check if it's persisted just to call save (compare w/ v1)
-            power_of_attorney = ClaimsApi::PowerOfAttorney.create!(attributes)
-          end
-
-          ClaimsApi::PoaUpdater.perform_async(power_of_attorney.id)
+        def submit_power_of_attorney(poa_code, form_number)
+          attributes = {
+            status: ClaimsApi::PowerOfAttorney::PENDING,
+            auth_headers:,
+            form_data: form_attributes,
+            current_poa: current_poa_code,
+            header_md5:
+          }
+          attributes.merge!({ source_data: }) unless token.client_credentials_token?
 
-          ClaimsApi::VBMSUpdater.perform_async(power_of_attorney.id) if enable_vbms_access?
+          power_of_attorney = ClaimsApi::PowerOfAttorney.create!(attributes)
 
           # This builds the POA form *AND* uploads it to VBMS
-          ClaimsApi::PoaFormBuilderJob.perform_async(power_of_attorney.id)
+          ClaimsApi::PoaFormBuilderJob.perform_async(power_of_attorney.id, form_number)
 
           render json: ClaimsApi::V2::Blueprints::PowerOfAttorneyBlueprint.render(
-            representative(poa_code).merge({ code: poa_code })
+            representative(poa_code).merge({ id: power_of_attorney.id, code: poa_code }),
+            root: :data
+          ), status: :accepted, location: url_for(
+            controller: 'power_of_attorney', action: 'show', id: power_of_attorney.id
           )
         end
 
         def validate2122a
-          target_veteran
-          validate_json_schema('2122a'.upcase)
-
-          poa_code = form_attributes.dig('representative', 'poaCode')
-          validate_individual_poa_code!(poa_code)
+          shared_form_validation('2122A')
 
           render json: validation_success
         end
 
         private
 
+        def shared_form_validation(form_number)
+          target_veteran
+          validate_json_schema(form_number.upcase)
+          poa_code = get_poa_code(form_number)
+          validate_individual_poa_code!(poa_code)
+          parse_and_validate_poa_code('2122A')
```

Comment:
I'm not sure `parse_and_validate_poa_code('2122A')` is needed since it repeats the `::Veteran::Service::Representative.where('? = ANY(poa_codes)', poa_code).any?` check done in `validate_individual_poa_code!`. The `validate_poa_code_for_current_user!` will get skipped in ccg flow.